applications/app/controllers/OptInController.scala

```diff
@@ -6,16 +6,28 @@ import play.api.mvc.{Action, Controller, Cookie, DiscardingCookie}
 
 import scala.concurrent.duration._
 
-object OptInController extends Controller {
-  case class OptInFeature(cookieName: String, lifetime: Int = 90.days.toSeconds.toInt) {
-    def opt(choice: String) = choice match {
-      case "in" => optIn()
-      case _ => optOut()
-    }
-
-    def optIn() = SeeOther("/").withCookies(Cookie(cookieName, "true", maxAge = Some(lifetime)))
-    def optOut() = SeeOther("/").discardingCookies(DiscardingCookie(cookieName))
+trait OptInOutFeature extends Controller {
```

Comment:
not 100% sure it was the right thing to add a trait like that and subclasses
the code might have remaned simpler if we just treated it as opting in to http only, from a coding point of view?